monkey/infection_monkey/monkey.py

```diff
@@ -180,14 +186,27 @@ def _setup(self):
         control_channel.register_agent(self._opts.parent)
 
         config = control_channel.get_config()
-        self._monkey_inbound_tunnel = self._control_client.create_control_tunnel(
-            config.keep_tunnel_open_time
+
+        local_port = get_free_tcp_port()
+        sock, ip_str, port = connect([self._opts.server])
+        sock.close()
```

Comment:
Are we doing this to make sure the port is actually free? `get_free_tcp_port()` will only return available/free ports.

Reply:
I was just keeping the status quo -- that's what MonkeyTunnel was using as the local port:
https://github.com/guardicore/monkey/blob/develop/monkey/infection_monkey/tunnel.py#L138

Reply:
Sure, but why connect and close?

Reply:
`connect()` uses a list of servers and returns connection info on the first one that it can connect to. It needs to establish a connection in order to verify that it is viable. I agree that having to close the returned socket is kinda wonky. I could wrap this in another function, `get_viable_connection()` that would just return the IP and port